allennlp/common/detectron.py

```diff
@@ -269,14 +285,245 @@ def get_pipeline_from_flat_parameters(
                 "TRAIN_ON_PRED_BOXES": fp.box_head_train_on_pred_boxes,
                 "BBOX_LOSS_WEIGHT": fp.box_head_bbox_loss_weight,
             },
-
             "ROI_ATTRIBUTE_HEAD": {  # not in detectron2 default config
                 "OBJ_EMBED_DIM": fp.attribute_head_obj_embed_dim,
                 "FC_DIM": fp.attribute_head_fc_dim,
                 "LOSS_WEIGHT": fp.attribute_head_loss_weight,
                 "NUM_CLASSES": fp.attribute_head_num_classes,
-            }
+            },
         },
+        "TEST": {"DETECTIONS_PER_IMAGE": fp.test_detections_per_image},
     }
 
     return get_pipeline(None, None, overrides, make_copy)
+
+
+class AttributePredictor(nn.Module):
```

Comment:
@jiasenlu, these classes are some extra stuff that we wish was part of Detectron, but isn't, correct? It has nothing directly to do with AllenNLP?

Reply:
yes. It has nothing directly to do with AllenNLP, but current detectron2 feature extraction model requires   "AttributeRes5ROIHeads", which we register this class blow. 

https://github.com/allenai/allennlp/pull/4481/files/8515be27a00aa1aebd3363630c89ad8c3d364535..5db6805b50fb1f0197c82daea2469687d5dc248a#diff-b2b05f69be08bf21366facbbe9abfc5dR174

Reply:
yes. It has nothing directly to do with AllenNLP, but current detectron2 feature extraction model requires "AttributeRes5ROIHeads", which we register this class blow.

https://github.com/allenai/allennlp/pull/4481/files/8515be27a00aa1aebd3363630c89ad8c3d364535..5db6805b50fb1f0197c82daea2469687d5dc248a#diff-b2b05f69be08bf21366facbbe9abfc5dR174